src/core_modules/capture-core/metaDataStoreLoaders/programs/loadPrograms.js

```diff
@@ -1,128 +1,56 @@
 // @flow
-import StorageController from 'capture-core-utils/storage/StorageController';
-
 import { chunk } from 'capture-core-utils';
-
-import metaProgramsSpec from '../../api/apiSpecifications/metaPrograms.apiSpecification';
-import programsSpec from '../../api/apiSpecifications/programs.apiSpecification';
-import programRulesSpec from '../../api/apiSpecifications/programRules.apiSpecification';
-import programRulesVariablesSpec from '../../api/apiSpecifications/programRulesVariables.apiSpecification';
-import programIndicatorsSpec from '../../api/apiSpecifications/programIndicators.apiSpecification';
-
-import getProgramsLoadSpecification from '../../apiToStore/loadSpecifications/getProgramsLoadSpecification';
-import getProgramRulesLoadSpecification from '../../apiToStore/loadSpecifications/getProgramRulesLoadSpecification';
-import getProgramRulesVariablesLoadSpecification
-    from '../../apiToStore/loadSpecifications/getProgramRulesVariablesLoadSpecification';
-import getProgramIndicatorsLoadSpecification
-    from '../../apiToStore/loadSpecifications/getProgramIndicatorsLoadSpecification';
-
-import programsStoresKeys from './programsStoresKeys';
-
-const batchSize = 50;
-
-async function getMissingPrograms(programs, storageController: StorageController, store: string) {
-    if (!programs) {
-        return null;
-    }
-
-    const storePrograms = {};
+import { queryProgramsOutline } from './queries';
+import {
+    storePrograms,
+    storeProgramRules,
+    storeProgramRulesVariables,
+    storeProgramIndicators,
+} from './quickStoreOperations';
+import { getContext } from '../context';
+
+async function getMissingPrograms(programsOutline) {
```

Comment:
Maybe I miss-understand something here what is the difference between `programs` and `programsOutline`?

Reply:
programs contains the full program object, programsOutline just parts of it.

Reply:
Thanks for that!